x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/RestEsqlIT.java

```diff
@@ -52,6 +52,50 @@ public void testBasicEsql() throws IOException {
         assertEquals(List.of(List.of(499.5d)), result.get("values"));
     }
 
+    public void testTimeout() throws IOException {
+        StringBuilder b = new StringBuilder();
+        for (int i = 0; i < 10000; i++) {
+            b.append(String.format(Locale.ROOT, """
+                {"create":{"_index":"%s"}}
+                {"test":"value%s"}
+                """, testIndexName(), randomAlphaOfLength(20) + " " + randomAlphaOfLength(20) + " " + randomAlphaOfLength(20)));
+        }
+        Request bulk = new Request("POST", "/_bulk");
+        bulk.addParameter("refresh", "true");
+        bulk.addParameter("filter_path", "errors");
+        bulk.setJsonEntity(b.toString());
+        Response response = client().performRequest(bulk);
+        Assert.assertEquals("{\"errors\":false}", EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8));
+
+        // let's make it a bit expensive, so that the timeout will most likely trigger
```

Comment:
We should create a test plugin to avoid the flakiness here.

Reply:
I was considering to add a `sleep(x)` function only in test, to simulate slow queries. Not sure it's what you mean, if you think there is a better way please let me know